editor/group_settings_editor.cpp

```diff
@@ -0,0 +1,531 @@
+	undo_redo->add_undo_property(ProjectSettings::get_singleton(), name, Variant());
+
+	undo_redo->add_do_method(this, "call_deferred", "update_groups");
+	undo_redo->add_undo_method(this, "call_deferred", "update_groups");
+
+	undo_redo->add_do_method(this, "emit_signal", group_changed);
+	undo_redo->add_undo_method(this, "emit_signal", group_changed);
+
+	undo_redo->commit_action();
+
+	group_name->clear();
+	group_description->clear();
+}
+
+void GroupSettingsEditor::_group_name_text_changed(const String &p_name) {
+	String error = _check_new_group_name(p_name.strip_edges());
+	add_button->set_tooltip_text(error);
+	add_button->set_disabled(!error.is_empty());
+}
+
+void GroupSettingsEditor::_modify_references(const StringName &p_name, const StringName &p_new_name, bool p_is_rename) {
+	HashSet<String> scenes;
+
+	HashMap<StringName, HashSet<StringName>> scene_groups_cache = ProjectSettings::get_singleton()->get_scene_groups_cache();
+	for (const KeyValue<StringName, HashSet<StringName>> &E : scene_groups_cache) {
+		if (E.value.has(p_name)) {
+			scenes.insert(E.key);
+		}
+	}
+
+	int steps = scenes.size();
+	Vector<EditorData::EditedScene> edited_scenes = EditorNode::get_editor_data().get_edited_scenes();
+	for (const EditorData::EditedScene &es : edited_scenes) {
+		if (!es.root) {
+			continue;
+		}
+		if (es.path.is_empty()) {
+			++steps;
+		} else if (!scenes.has(es.path)) {
+			++steps;
+		}
+	}
+
+	String progress_task = p_is_rename ? "rename_reference" : "remove_references";
+	String progress_label = p_is_rename ? TTR("Renaming Group References") : TTR("Removing Group References");
+	EditorProgress progress(progress_task, progress_label, steps);
+
+	int step = 0;
+	// Update opened scenes.
+	HashSet<String> edited_scenes_path;
+	for (const EditorData::EditedScene &es : edited_scenes) {
+		if (!es.root) {
+			continue;
+		}
+		progress.step(es.path, step++);
+		bool edited = p_is_rename ? rename_node_references(es.root, p_name, p_new_name) : remove_node_references(es.root, p_name);
+		if (!es.path.is_empty()) {
+			scenes.erase(es.path);
+			if (edited) {
+				edited_scenes_path.insert(es.path);
+			}
+		}
+	}
+	if (!edited_scenes_path.is_empty()) {
+		EditorNode::get_singleton()->save_scene_list(edited_scenes_path);
+		SceneTreeDock::get_singleton()->get_tree_editor()->update_tree();
+	}
+
+	for (const String &E : scenes) {
+		Ref<PackedScene> packed_scene = ResourceLoader::load(E);
+		progress.step(E, step++);
+		ERR_CONTINUE(packed_scene.is_null());
+		if (p_is_rename) {
+			if (packed_scene->get_state()->rename_group_references(p_name, p_new_name)) {
+				ResourceSaver::save(packed_scene, E);
+			}
+		} else {
+			if (packed_scene->get_state()->remove_group_references(p_name)) {
+				ResourceSaver::save(packed_scene, E);
+			}
+		}
+	}
+}
+
+void GroupSettingsEditor::remove_references(const StringName &p_name) {
+	_modify_references(p_name, "", false);
```

Comment:
```suggestion
	_modify_references(p_name, StringName(), false);
```
Not sure how much it matters, might be optimized anyway.